src/widget/colormenu.cpp

```diff
@@ -1,31 +1,51 @@
 #include "widget/colormenu.h"
+
+#include <QGridLayout>
+#include <QMapIterator>
+#include <QPushButton>
+
 #include "util/color/color.h"
 
-ColorMenu::ColorMenu(QWidget *parent)
-        : QMenu(parent) {
+namespace {
+const int kNumColumns = 4;
+}
+
+ColorMenu::ColorMenu(QWidget* parent)
+        : QWidget(parent) {
     // If another title would be more appropriate in some context, setTitle
     // can be called again after construction.
-    setTitle(tr("Set color"));
+    QGridLayout* pLayout = new QGridLayout();
+    pLayout->setMargin(0);
+    pLayout->setContentsMargins(0,0,0,0);
+    int row = 0;
+    int column = 0;
     for (const auto& pColor : Color::kPredefinedColorsSet.allColors) {
         if (*pColor == *Color::kPredefinedColorsSet.noColor) {
             continue;
         }
 
-        QAction* pColorAction = new QAction(pColor->m_sDisplayName, this);
-        QPixmap pixmap(80, 80);
-        pixmap.fill(pColor->m_defaultRgba);
-        pColorAction->setIcon(QIcon(pixmap));
+        QPushButton* pColorButton = new QPushButton("", this);
+        QPalette p(pColorButton->palette());
```

Comment:
we prefer:  
QPalette palette = pColorButton->palette()
because it reads more natural and produces the same code.

Reply:
p is our pointer prefix.